Algorithm.Python/TensorFlowNeuralNetworkAlgorithm.py

```diff
@@ -0,0 +1,112 @@
+# QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+# Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License"); 
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import clr
+clr.AddReference("System")
+clr.AddReference("QuantConnect.Algorithm")
+clr.AddReference("QuantConnect.Common")
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Algorithm import *
+
+
+import numpy as np
+import tensorflow as tf
+
+class TensorFlowNeuralNetworkAlgorithm(QCAlgorithm):
```

Comment:
1) Please add comments to each algorithm class about what it is doing. 
2) Please aggressively comment on ALL parts of the algorithm. These algorithms are _documentation_ so important to comment them well.